util/azure/utils.go

```diff
@@ -162,39 +189,69 @@ func ConvertIntToString(number int) string {
+	if settings == nil {
+		settings = &DiscoverResourcesSettings{
+			clusterType:        clusterType,
+			aksLoginURL:        loginURL,
+			kubeconfigFilePath: kubeconfigFilePath,
+			tenantID:           tenantID,
+			clientID:           clientID,
+			clientSecret:       clientSecret,
+		}
 
-	env := azure.PublicCloud
-	var err error
-	if environment != "" {
-		env, err = azure.EnvironmentFromName(environment)
-		if err != nil {
-			return nil, errors.Wrap(err, "Failed to parse environment for Azure.")
+		env := azure.PublicCloud
+		var err error
+		if environment != "" {
+			env, err = azure.EnvironmentFromName(environment)
+			if err != nil {
+				return errors.Wrap(err, "Failed to parse environment for Azure.")
+			}
 		}
-	}
 
-	settings.environment = env
-
-	switch clusterType {
-	case ConnectedClusters:
-		settings.operationsEndpoint = fmt.Sprintf(OperationsEndpointFormatARC, settings.environment.ResourceManagerEndpoint)
-	case ManagedClusters:
-		settings.operationsEndpoint = fmt.Sprintf(OperationsEndpointFormatAKS, settings.environment.ResourceManagerEndpoint)
-	case Fleets:
-		settings.operationsEndpoint = fmt.Sprintf(OperationsEndpointFormatAKS, settings.environment.ResourceManagerEndpoint)
-	default:
-		return nil, errors.Errorf("Failed to create endpoint for Get Operations call. Cluster type %s is not supported.", clusterType)
+		settings.environment = env
+
+		switch clusterType {
+		case ConnectedClusters:
+			settings.operationsEndpoint = fmt.Sprintf(OperationsEndpointFormatARC, settings.environment.ResourceManagerEndpoint)
+		case ManagedClusters:
+			settings.operationsEndpoint = fmt.Sprintf(OperationsEndpointFormatAKS, settings.environment.ResourceManagerEndpoint)
+		case Fleets:
+			settings.operationsEndpoint = fmt.Sprintf(OperationsEndpointFormatAKS, settings.environment.ResourceManagerEndpoint)
+		default:
+			return errors.Errorf("Failed to create endpoint for Get Operations call. Cluster type %s is not supported.", clusterType)
+		}
+
+		return nil
 	}
+	return nil
+}
 
-	return settings, nil
+// ReconcileDiscoverResources reconciles the GlobalOperationsMap
+func ReconcileDiscoverResources(ctx context.Context, wg *sync.WaitGroup, loopDuration time.Duration) {
+	defer wg.Done()
+	for {
+		select {
+		case <-time.After(loopDuration):
+			discoverResourcesListStart := time.Now()
+			err := DiscoverResources()
+			discoverResourcesDuration := time.Since(discoverResourcesListStart).Seconds()
+			if err != nil {
+				code := http.StatusInternalServerError
+				if v, ok := err.(errutils.HttpStatusCode); ok {
+					code = v.Code()
+				}
+				counterDiscoverResources.WithLabelValues(ConvertIntToString(code)).Inc()
+				DiscoverResourcesTotalDuration.Observe(discoverResourcesDuration)
+				klog.Errorf("Failed to reconcile map of data actions. Error:%s", err)
```

Comment:
you should return here?

Reply:
since it's a reconciling loop, i thought we should not error out on error but instead just log it. hence not returning here

Reply:
on startup we are throwing any non apiservice error as fatal but during reconciling since this could be a transient error and we already would have a old version of the map that can be used we don't need to fatalf here but we can just show error here.